File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& nums) {
        vector<int> l;
        vector<int> r;
        int ll=0, rr=0;
        for(auto it:nums){
            ll = max(ll, it);
            l.push_back(ll);
        }
        for (int it=nums.size()-1; it>=0; --it) {
         
            rr = max(rr, nums[it])   ;
            r.push_back(rr);
        }

        int le=0, ri=nums.size()-1;
        int ans=0;
        while(le<ri){
            int h = min(nums[le], nums[ri]);
            ans = max(ans, h*(ri-le));
            if(nums[le] < nums[ri]) le++;
            else ri--;
        }
        return ans;

    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.java
class Solution {
    public int longestConsecutive(int[] nums) {
        int ans = 0, cnt = 1, last_smaller = Integer.MIN_VALUE;
        if (nums.length <= 1)
            return nums.length;

        Arrays.sort(nums);

        for (int i = 1; i < nums.length; i++) {
            if (nums[i] - 1 == last_smaller) {
                cnt+=1;
                last_smaller = nums[i];
                System.out.println(nums[i]);
            } else if (nums[i] != last_smaller) {
                cnt = 1;
                last_smaller = nums[i];
            }
            ans = Math.max(ans, cnt);
        }
        return ans+1;
    }
}
